File: app/src/main/java/com/example/ls/maillistdemo/FileUtil.java
package com.example.ls.maillistdemo;

import android.content.Context;

import java.io.BufferedReader;
import java.io.InputStream;
import java.io.InputStreamReader;

/**
 * Created by LS on 2018/1/9.
 */

public class FileUtil {


    /**
     * 从assets中读取txt
     */
    public static String readFromAssets(Context context, String fileName) {
        String text = "";
        synchronized (FileUtil.class) {
            try {
                InputStream is = context.getAssets().open(fileName);
                text = readTextFromSDcard(is);
            } catch (Exception e) {
                e.printStackTrace();
            }
            return text;
        }
    }

    /**
     * 按行读取txt
     *
     * @param is
     * @return
     * @throws Exception
     */
    private static String readTextFromSDcard(InputStream is) throws Exception {
        InputStreamReader reader = new InputStreamReader(is);
        BufferedReader bufferedReader = new BufferedReader(reader);
        StringBuffer buffer = new StringBuffer("");
        String str;
        while ((str = bufferedReader.readLine()) != null) {
            buffer.append(str);
            buffer.append("\n");
        }
        return buffer.toString();
    }

}

File: app/src/main/java/com/example/ls/maillistdemo/FileUtil.kt
package com.example.ls.maillistdemo

import android.content.Context
import java.io.BufferedReader
import java.io.InputStream
import java.io.InputStreamReader

/**
 * Created by LS on 2018/1/9.
 */

object FileUtil {


    /**
     * 从assets中读取txt
     */
    fun readFromAssets(context: Context, fileName: String): String {
        var text = ""
        synchronized(FileUtil::class.java) {
            try {
                val content = context.assets.open(fileName)
                text = readTextFromSDcard(content)
            } catch (e: Exception) {
                e.printStackTrace()
            }

            return text
        }
    }

    /**
     * 按行读取txt
     *
     * @param is
     * @return
     * @throws Exception
     */
    @Throws(Exception::class)
    private fun readTextFromSDcard(content: InputStream): String {
        val reader = InputStreamReader(content)
        var bufferedReader = BufferedReader(reader)
        val buffer = StringBuffer("")
        var str: String
        str = bufferedReader.readLine()
        try {
            while (str != null) {
                buffer.append(str)
                buffer.append("\n")
                str = bufferedReader.readLine()
            }
        } catch (e: Exception) {
        } finally {
        }
        return buffer.toString()
    }

}